src/test/java/com/opipo/codechallenge/it/steps/TransactionSteps.java

```diff
@@ -0,0 +1,257 @@
+package com.opipo.codechallenge.it.steps;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.opipo.codechallenge.repository.TransactionRepository;
+import com.opipo.codechallenge.repository.model.TransactionEntity;
+import com.opipo.web.api.model.Transaction;
+import com.opipo.web.api.model.TransactionTesting;
+
+import io.cucumber.java.Before;
+import io.cucumber.java8.En;
+
+public class TransactionSteps implements En {
+    private static final String ENDPOINT = "http://localhost:8080/transaction";
+
+    @Autowired
+    private TransactionRepository transactionRepository;
+
+    private RestTemplate restTemplate;
+
+    private Transaction transaction;
+
+    private TransactionEntity transactionStored;
+
+    private ResponseEntity<Void> createResponse;
+
+    private ResponseEntity<Transaction[]> listResponse;
+
+    private Collection<Transaction> transactionsPersisted = new ArrayList<>();
+
+    private HttpStatus errorStatus;
+
+    private Random random = new Random();
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    @Before
+    public void init() {
+        this.restTemplate = new RestTemplate();
+    }
+
+    public TransactionSteps() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The constructor 'TransactionSteps()' has an NPath complexity of 196608](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)